api/app/src/main/kotlin/packit/service/RoleService.kt

```diff
@@ -94,10 +95,9 @@ class BaseRoleService(
         val role = roleRepository.findByName(roleName)
             ?: throw PackitException("roleNotFound", HttpStatus.BAD_REQUEST)
```

Comment:
I think we should just remove this wrapper from this class and have the controller call `RolePermissionService.updatePermissionsOnRole` directly.

The `RoleController` calls `UserRoleService.updateRoleUsers` directly for example.

Reply:
i feel it just splits the logic up better... preprocess any role related behaviour in the role service, so the rolepermission service can just deal with their relationship